src/main/java/com/flagsmith/FlagsmithCacheConfig.java

```diff
@@ -44,35 +46,49 @@ private FlagsmithCacheConfig(Builder builder) {
 
   class FlagsmithInternalCache implements FlagsmithCache {
     private final Cache<String, FlagsAndTraits> cache;
+    private final String projectFlagsCacheKey;
 
-    public FlagsmithInternalCache(Cache<String, FlagsAndTraits> cache) {
```

Comment:
Should we have an overload on this that doesn't require the cache key and just defaults it to some random string or maybe an environment variable? Just concerned that this change is not backwards compatible, right?

Reply:
Overloaded constructor added.